### DISCUSSION TITLE
A slightly different use case: TV on-screen-display

### DISCUSSION THREAD

**Discussion by harbaum** [olikraus/u8g2#2288] [Show and tell]
I am building a FPGA based retro computer (see http://GitHub.com/harbaum/MiSTeryNano) and these typically need an on-screen-display (OSD) for basic configuration.

To save memory in the FPGA, these are usually pretty low-rez and monochrome. Previously we were doing simple text based menus but I thought I'd give u8g2 a try and to my surprise found the mui. So I gave that a try. And this is how it looks.

![IMG_20231029_173002.jpg](https://github.com/olikraus/u8g2/assets/9105084/19ad3027-7153-43dd-8740-5f215ca5a106)

**Comment by harbaum** [comment]
Interesting fact: The original OSD had 128x64 pixels and the bytes were oriented for simple text display.  Yes, the original OSD had exactly the same memory layout as the typical mini OLED display. Unfortunately I had already migrated to a regular framebuffer layout with 'horizontal bytes' so I had to revert to the 'vertical bytes'.

The OSD is a simple framebuffer inside the FPGA overlayed to the video whenever needed. It's made available to the external BL616 microcontroller via SPI. This microcontroller also does USB host to control mouse and keyboard which are used to control the embedded retro machine as well as the OSD.

One major job for the OSD will be floppy disk image selection from an SD card. So has someone perhaps already done a file browser for mui?

**Reply by olikraus** [reply]
Very nice idea. Cool project 👍. And no, 8 have not added a file browser yet.
